src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -503,6 +502,80 @@ public void completeTask(String taskId, AccessControlResponse accessControlRespo
         }
     }
 
+    private void checkPermissions(String taskId, AccessControlResponse accessControlResponse,
+            boolean isGranularPermissionFeatureEnabled, String userId) {
+        PermissionRequirements permissionsRequired;
+        if (isGranularPermissionFeatureEnabled) {
+            permissionsRequired = PermissionRequirementBuilder.builder()
```

Comment:
Isn't it possible pass all the permission requirements at once and then do the additional checks base on the TaskResource returned by the verifyRoleAssignments method? In this case, if the verification returns task resource with TaskRoleResource with own or execute flag is true, or completeown flag is true, then check the assignee. Similar to removed line no 455?